api/lib/firewall_functions.pl

```diff
@@ -324,13 +327,38 @@ sub list_fw_rules
         $props{'Time'} = get_time_info($props{'Time'}, $fw->{'ftdb'}, $expand);
         $props{'Service'} = get_service_info($props{'Service'}, $fw, $expand);
         $props{'Position'} = int($props{'Position'});
-        $max_pos = max($max_pos, $props{'Position'});
+        $max_posRule = max($max_posRule, $props{'Position'});
         $nextID =`/usr/libexec/nethserver/api/nethserver-firewall-base/lib/rules-next-id`;
         push(@rules, \%props);
         $i++;
     }
+    foreach ($fw->getSeparator()) {
+        my %props = $_->props;
+        if ($skip_local) {
+            # skip rule to/from the firewall
+            next if ($props{'Dst'} eq 'fw');
+        } else {
+            # skip rule not for the firewall itself
+            next if ($props{'Dst'} ne 'fw');
+        }
+        $props{'id'} = $_->key;
+        $props{'Position'} = int($props{'Position'});
+        $nextID =`/usr/libexec/nethserver/api/nethserver-firewall-base/lib/rules-next-id`;
+        $max_posSeparator = max($max_posSeparator, $props{'Position'});
+        push(@separators, \%props);
+        $i++;
+    }
+    # find the the higher position in the fwrule or separator db 
+    if ($max_posSeparator >= $max_posRule ) {
+        $max_pos = $max_posSeparator;
+    } else {
+        $max_pos = $max_posRule;
+    }
```

Comment:
The rules-next-id should do already the same job :thinking:

Reply:
The rules-next-id parse all keys of the two databases and output the next id available (the higher of the two databases), the rule-next-position is used only when you create a new rule because we drop by default at the end of the list, so we check all positions of all rules/separators to find the last one and we increment of one to put the rule to the last position.

For memory a separator is created at the top of the list.

Reply:
Yes, and this code does the same.
We can greatly simplify the whole function, I can't produce a diff for Github but this is the simplified version:
```perl
sub list_fw_rules
{
    my $expand = shift;
    my $skip_local = shift;
    my $fw = new NethServer::Firewall();
    my @rules;
    my $max_pos = 0;
    my $max_posRule = 0;
    my $max_posSeparator = 0;
    my $next = 0;
    foreach ($fw->getRules()) {
        my %props = $_->props;

        if ($skip_local) {
            # skip rule to/from the firewall
            next if ($props{'Src'} eq 'fw' or $props{'Dst'} eq 'fw');
        } else {
            # skip rule not for the firewall itself
            next if ($props{'Src'} ne 'fw' and $props{'Dst'} ne 'fw');
        }

        $props{'id'} = $_->key;

        $props{'Src'} = get_target_info($props{'Src'}, $fw, $expand);
        $props{'Dst'} = get_target_info($props{'Dst'}, $fw, $expand);
        $props{'Time'} = get_time_info($props{'Time'}, $fw->{'ftdb'}, $expand);
        $props{'Service'} = get_service_info($props{'Service'}, $fw, $expand);
        $props{'Position'} = int($props{'Position'});
        $max_posRule = max($max_posRule, $props{'Position'});
        push(@rules, \%props);
    }
    foreach ($fw->getSeparators()) {
        my %props = $_->props;
        $props{'id'} = $_->key;
        $props{'Position'} = int($props{'Position'});
        $max_posSeparator = max($max_posSeparator, $props{'Position'});
        push(@rules, \%props);
    }
    $next = int(`/usr/libexec/nethserver/api/nethserver-firewall-base/lib/rules-next-id`);

    my @sorted = sort { $a->{'Position'} <=> $b->{'Position'} } @rules;

    return {"status" => {"count" => scalar(@rules), "nextID" => $next }, "rules" => \@sorted};
}

```

Reply:
not sure to understand but the UI expects to know the next position where to create a rule see the missing `next` in the return of your proposal: `"next" => $max_pos+1`

-    return {"status" => {"count" => scalar(@rules), "next" => $max_pos+1, "nextID" => $nextID }, "rules" => \@sorted};

+    return {"status" => {"count" => scalar(@rules), "nextID" => $next }, "rules" => \@sorted};

Reply:
The UI seems to use only the nextID option. Is part of one your last commits on this part to reorder the rules only from the web interface